src/components/application_manager/src/commands/mobile/register_app_interface_request.cc

```diff
@@ -816,12 +816,12 @@ void RegisterAppInterfaceRequest::SendRegisterAppInterfaceResponseToMobile() {
                                           &(msg_params[strings::app_hmi_type]));
   }
 #endif  // SDL_REMOTE_CONTROL
+  SendResponse(true, result_code, add_info.c_str(), &response_params);
+
   // Default HMI level should be set before any permissions validation, since it
   // relies on HMI level.
   application_manager_.OnApplicationRegistered(application);
 
-  SendResponse(true, result_code, add_info.c_str(), &response_params);
```

Comment:
@LuxoftAKutsan BC.ActivateApp is sent by PolicyHandler::UpdateHMILevel(...) which is called by PolicyHandler::OnUpdateHMILevel(...), triggered by the change of the HMILevel in ApplicationManagerImpl::OnApplicationRegistered(...) --> line 823.
OnAppRegistered is sent by RegisterAppInterfaceRequest::SendOnAppRegisteredNotificationToHMI(...) --> line 811.
As a result BC.ActivateApp is sent after OnAppRegistered.